pkg/apis/core/v1alpha1/defaults_test.go

```diff
@@ -581,19 +581,19 @@ var _ = Describe("Defaults", func() {
 		It("should set the maintenance field", func() {
 			obj.Spec.Maintenance = nil
 
-			SetDefaults_Shoot(obj)
+			SetObjectDefaults_Shoot(obj)
 
-			Expect(obj.Spec.Maintenance).To(Equal(&Maintenance{}))
+			Expect(obj.Spec.Maintenance).NotTo(BeNil())
```

Comment:
Is there any reason to do this change? Theoretically, one could adapt the defaulting code that a maintenance obj is populated with some fields. The adapted assertion from this commit won't catch such thing.

Reply:
Before this commit, only `SetDefaults_Shoot` was called which adds an empty maintenance section (`SetDefaults_Maintenance` was not invoked in the test).

Only the full `SetObjectDefaults_Shoot` (which is the actual defaulting func registered in the scheme) invokes `SetDefaults_Maintenance`, which also defaults the fields of the maintenance section.

I don't see a point in asserting that the `SetDefaults_Shoot` adds an empty maintenance section exactly (this was tested before), because this will never be called standalone in the actual production code of the API server.
In my opinion, the cohesive unit that should be tested in these unit tests is the actual defaulting func (`SetObjectDefaults_Shoot`) that will also be called in the API server.
`SetDefaults_Shoot` is only an "implementation detail" of the unit and should not be tested standalone.

We could drop the `should set the maintenance field` test case and rather test the entire maintenance section defaulting together. This test case would already be covered by the `should correctly default the maintenance` case.